### PR TITLE
[UG] Add annotated GUI info and format student fields table

### LINKED ISSUE #231: [PE-D] Target user for the product is not specified clearly
The target user is not specified since it is unclear who "you" refers to.

Perhaps you may rephrase it to "private tutors"

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: ganhongyao/ped#1

### LINKED ISSUE #237: [PE-D] Students with the same name are not allowed
Steps to reproduce
1. open app
2. execute command `add n/Ryan Tan a/moony drive e/tan@gmail.com`
3. execute command `add n/Ryan Tan a/sunny drive e/kow@gmail.com`

Expected behaviour
- Allow student to be added since although they have the same name, they may be different students.

Actual behaviour
- "This student already exists in TAB!"

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: ganhongyao/ped#3

### LINKED ISSUE #255: [PE-D] Vague detail provided on UI
It may be confusing to the user what the "Date" field means, does it represent the start or the end date for classes that are recurring. Perhaps it could be better labelled?

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: pattan9955/ped#6

### LINKED ISSUE #323: UG image not updated
This is not what I see when I load TAB
image

What I see is this. I think the image can be updated to prevent the testers from nitpicking
image

### LINKED ISSUE #324: Help image is too small
If diagrams in the DG are too small is considered a bug, this is probs a bug too
image

### PR DESCRIPTION
**Pull request by Xiaoyunnn**
Fixes #231 
Fixes #237 
Fixes #255 // actually i feel that the desc table for the lesson card on GUI is a bit repetitive (similar to the lesson fields table in Eliana's PR), but it clarifies wat the fields mean on GUI, particularly for the date range, date, cancelled dates 
Fixes #323 
Fixes #324

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -34,9 +35,9 @@ Syntax | Meaning
 <kbd>text</kbd> | A keyboard input, or a button to be clicked on.
 [text](#about-this-guide) | Links to other parts of the document, or links to be opened in the browser.
 _text_ | Caption for images.
-:information_source: | Indication that the following text is a note.
-:bulb: | Indication that the following text is a tip.
-:exclamation: | Indication that the following text is important.
+<div markdown="block" class="alert alert-info"> :information_source: **Note** </div> | Indication that the following text is a note, which is useful in helping you understand how TAB works.
+<div markdown="span" class="alert alert-primary"> :bulb: **Tip** </div> | Indication that the following text is a tip, which can allows you to interact with TAB more effectively and quickly.
```
Comment (eeliana) [inline docs/UserGuide.md:39]:
```suggestion
<div markdown="span" class="alert alert-primary"> :bulb: **Tip** </div> | Indication that the following text is a tip, which can allow you to interact with TAB more effectively and quickly. 
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -88,16 +89,22 @@ to help you with the installation. Follow the guide for your operating system fo
...
-2 | Result Display | The area that shows the result of the execution of the command. If the command entered has been executed successfully, it will display relevant success messages, otherwise, it will show error messages indicating the cause of the error.
-3 | Student List Panel | The area that shows the list of students you have in TAB.
-4 | Lesson List Panel | The area that shows the name of the student you have selected together with a list of lessons that the student has.
-5 | Center Panel | The area that displays the lists of students and lessons, calendar or the list of tags depending on the command you have entered. See [Features](#features) for more details.
+Section | Representation
+---------|---------
+Menu Bar | The area where you can click to navigate and view some of TAB's features (e.g. help, reminder and calendar). See [Features](#features) for more details.
+Menu Item | A button that you can click to view the stated feature, or to quit TAB (for <kbd>Exit</kbd>).
+Command Box | The text field where you key in commands that are meant to be processed by TAB. TAB will execute the command after you pressed <kbd>Enter</kbd>.
```
Comment (eeliana) [inline docs/UserGuide.md:99]:
```suggestion
Command Box | The text field where you key in commands that are meant to be processed by TAB. TAB will execute the command after you press <kbd>Enter</kbd>.
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -88,16 +89,22 @@ to help you with the installation. Follow the guide for your operating system fo
...
+Result Display | The area that shows the result of the execution of the command. If the command entered has been executed successfully, it will display relevant success messages, otherwise, it will show error messages indicating the cause of the error.
```
Comment (eeliana) [inline docs/UserGuide.md:100]:
```suggestion
Result Display | The area that shows the result of the execution of the command. If the command entered has been executed successfully, it will display the relevant success message. Otherwise, it will show an error message indicating the cause of the error.
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -88,16 +89,22 @@ to help you with the installation. Follow the guide for your operating system fo
...
+Student List | The area that shows the list of students you have in TAB.
+Student Card | The area that shows the details of a student. See [Managing Students](#managing-students) for more details.
```
Comment (eeliana) [inline docs/UserGuide.md:102]: Area for card sounds strange but I don't know what could be a better alternative

File: docs/UserGuide.md [hunk 5]
```diff
@@ -88,16 +89,22 @@ to help you with the installation. Follow the guide for your operating system fo
...
+Menu Item | A button that you can click to view the stated feature, or to quit TAB (for <kbd>Exit</kbd>). <br> - Note: Clicking on <kbd>View</kbd> → <kbd>Students</kbd> brings you to the last displayed list of students and not the full list of students.
+Command Box | The text field where you key in commands that are meant to be processed by TAB. TAB will execute the command after you pressed <kbd>Enter</kbd>.
+Result Display | The area that shows the result of the execution of the command. If the command entered has been executed successfully, it will display relevant success messages, otherwise, it will show error messages indicating the cause of the error.
+Student List | The area that shows the list of students you have in TAB.
+Student Card | The area that shows the details of a student. See [Managing Students](#managing-students) for more details.
+Lesson List | The area that shows the name of the student you have selected together with a list of lessons that the student has.
+Lesson Card | The area that shows the details of a lesson. See [Managing Lessons](#managing-lessons) for more details.
```
Comment (eeliana) [inline docs/UserGuide.md:104]: Same comments for these.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -713,13 +815,13 @@ Format: `exit`
...
-<kbd>F3</kbd> | **View Calendar** <br /> Brings you to the calendar interface.
-<kbd>F4</kbd> | **View Tags** <br /> Brings you to your list of tags.
-<kbd>F5</kbd> | **View Reminders** <br /> Opens the reminder window.
+Shortcut | Action | Description
+--------|---------|---------
+<kbd>F1</kbd> | Help | Opens the help window.
+<kbd>F2</kbd> | View Students | Brings you to the last displayed list of students. <br> e.g. if you used `find` to filter the list of students, and switched to the calendar view, pressing <kbd>F2</kbd> brings you back to the filtered list.
+<kbd>F3</kbd> | View Calendar | Brings you to the calendar interface.
+<kbd>F4</kbd> | View Tags | Brings you to your list of tags.
+<kbd>F5</kbd> | View Reminder | Opens the reminder window with the list of upcoming lessons with end date time within the next 48 hours.
```
Comment (eeliana) [inline docs/UserGuide.md:824]: Not sure if end date time is used by non-technical person.
```suggestion
<kbd>F5</kbd> | View Reminder | Opens the reminder window with the list of upcoming lessons that ends within the next 48 hours.
```

**Comment by eeliana** [comment]
#323

**Comment by lingshanng** [comment]
> // actually i feel that the desc table for the lesson card on GUI is a bit repetitive (similar to the lesson fields table in Eliana's PR), but it clarifies how wat the fields mean on GUI, particularly for the date range, date, cancelled dates

Ya i think so too. What about instead of table, just more labelled pictures to explain date range, date and cancelled date

**Comment by Xiaoyunnn** [comment]
> > // actually i feel that the desc table for the lesson card on GUI is a bit repetitive (similar to the lesson fields table in Eliana's PR), but it clarifies how wat the fields mean on GUI, particularly for the date range, date, cancelled dates
> 
> Ya i think so too. What about instead of table, just more labelled pictures to explain date range, date and cancelled date

but the explanations are quite long also

**Comment by lingshanng** [comment]
I'm thinking just like this is sufficient?

Start Date<- Lesson start date
End Date <-Lesson end date which defaults to ∞ if not specified.
Date <- Date of the nearest upcoming lesson 

Makup Lesson picture
Date <- Date of the lesson
Cancelled! <- indicates that ...

**Comment by lingshanng** [comment]
Too squeezed
image

File: docs/UserGuide.md [hunk 7]
```diff
@@ -165,29 +182,197 @@ Format: `help`
...
 </div>
 
+<table id="student-param-table">
+    <thead>
+        <tr>
+            <th style="text-align:center; padding: 10px">Category</th>
+            <th style="text-align:center">Field</th>
+            <th style="text-align:center">Prefix</th>
+            <th style="text-align:center">Constraint(s)</th>
+            <th style="text-align:center">Example(s)</th>
```
Comment (lingshanng) [inline docs/UserGuide.md:204]: It's fine without
```suggestion
            <th style="text-align:center">Constraints</th>
            <th style="text-align:center">Examples</th>
```

File: docs/UserGuide.md [hunk 8]
```diff
@@ -165,29 +182,197 @@ Format: `help`
...
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan=2>Essential</td>
+            <td>Name</td>
+            <td><code>n/</code></td>
+            <td>
+              <ul> 
+                <li>Must be present.</li>
```
Comment (lingshanng) [inline docs/UserGuide.md:214]: It's inferred from essential
```suggestion
```

File: docs/UserGuide.md [hunk 9]
```diff
@@ -165,29 +182,197 @@ Format: `help`
...
+            <td><ul>
+              <li><code>n/Jane</code></li>
+              <li><code>n/Jane n/John</code> - name will be <code>John</code>.</li>
+            </ul></td>
+        </tr>
+        <tr>
+            <td>Address</td>
+            <td><code>a/</code></td>
+            <td><ul>
+              <li>Must be present.</li>
```
Comment (lingshanng) [inline docs/UserGuide.md:225]:
```suggestion
```

File: docs/UserGuide.md [hunk 10]
```diff
@@ -165,29 +182,197 @@ Format: `help`
...
+              <li><code>a/Blk 123A, Serangoon Ave 3, #04-56</code></li>
+              <li><code>a/Blk 123A, Serangoon Ave 3, #04-56 a/Clementi</code> - address will be <code>Clementi</code>.</li>
+            </ul></td>
+        </tr>
+        <tr>
+            <td rowspan=4 id="student-contact">Contact</td>
+            <td id="phone-field">Phone</td>
+            <td><code>p/</code></td>
+            <td><ul>
+              <li><strong>At least one</strong> of the contact fields must be present.</li>
```
Comment (lingshanng) [inline docs/UserGuide.md:234]: Add in the notes block above instead?

File: docs/UserGuide.md [hunk 11]
```diff
@@ -165,29 +182,197 @@ Format: `help`
...
+              <li><code>t/unpaid</code> adds only one tag with tag name "UNPAID"</li>
+              <li><code>t/unpaid retained</code> is invalid.</li>
+              <li><code>t/unpaid t/retained</code> adds two tags with tag names "UNPAID" and "RETAINED"</li>
+              <li><code>t/</code> clears <strong>all</strong> tags, if there were any.</li>
+            </ul></td>
+        </tr>
+        <tr>
+            <td rowspan="2">View Only</td>
+            <td>Student Index</td>
+            <td>Not Applicable.</td>
```
Comment (lingshanng) [inline docs/UserGuide.md:331]: no need full stop
```suggestion
            <td>Not Applicable</td>
```

File: docs/UserGuide.md [hunk 12]
```diff
@@ -200,9 +385,7 @@ Format: `add n/NAME a/ADDRESS [p/PHONE_NUMBER] [e/EMAIL] [pp/PARENT_PHONE_NUMBER
 A student can have any number of tags (including 0).
 </div>
 
-* At least one contact field is required.<br>
-  e.g. at least one of the `p/PHONE_NUMBER`, `e/EMAIL`, `pp/PARENT_PHONE_NUMBER`, or `pe/PARENT_EMAIL` fields must be 
-  included in the add command.
+* **At least one** [contact](#student-contact) field is required.<br>
```
Comment (lingshanng) [inline docs/UserGuide.md:358]: make the link [contact field]()

File: docs/UserGuide.md [hunk 13]
```diff
@@ -493,14 +685,38 @@ Examples:
...
 
+Example: Suppose the date today is 1 Nov 2021 and current time is 1500h, 
+- lessons with the following dates and time are considered upcoming:
+  - 1 Nov 2021 with end time at or after 1500h, 
+  - 2 Nov 2021 with any valid time range, 
+  - 3 Nov 2021 with start time before or at 1500h.
+- lessons with the following dates and time are not considered upcoming:
+  - dates before 1 Nov 2021 (has passed),
+  - 1 Nov 2021 with end time before 1500h (has passed),
+  - 3 Nov 2021 with start time after 1500 (beyond 48 hours).
```
Comment (lingshanng) [inline docs/UserGuide.md:675]:
```suggestion
  - 3 Nov 2021 with start time after 1500h (beyond 48 hours).
```

File: docs/UserGuide.md [hunk 14]
```diff
@@ -165,29 +182,167 @@ Format: `help`
...
+              <li><code>r/</code> clears the remark field, if there were any.</li>
+            </ul></td>
+        </tr>
+        <tr>
+            <td>Tags</td>
+            <td><code>t/</code></td>
+            <td><ul>
+              <li>Must be alphanumeric characters.</li>
+              <li>Case-insensitive</li>
+              <li>To add multiple tags to a student, you need to add <code>t/</code> before every tag name.</li>
```
Comment (lingshanng) [inline docs/UserGuide.md:319]: I think adding multiple tags should be address in ladd and not here

Comment (lingshanng) [reply docs/UserGuide.md:319]: Just need these examples I think
`t/unpaid` is a valid tag
`t/unpaid retained` is invalid

File: docs/UserGuide.md [hunk 15]
```diff
@@ -165,29 +182,167 @@ Format: `help`
...
+            <td>Name</td>
+            <td><code>n/</code></td>
+            <td>
+              <ul> 
+                <li>Must be present.</li>
+                <li>Case-sensitive.</li>
+                <li>TAB does not allow adding students with the same case-sensitive name.</li>
+              </ul>
+            </td>
+            <td><code>n/Jane</code></td>
```
Comment (lingshanng) [inline docs/UserGuide.md:219]:
```suggestion
            <td><code>n/Jane Tan</code></td>
```

File: docs/UserGuide.md (hunk 14 above)
Comment (lingshanng) [reply docs/UserGuide.md:311]: can remove the multiple tags one, since that depends on the command whether it accepts multiple prefixes

**Review by lingshanng — APPROVED** [review]
LGTM
